### DISCUSSION TITLE
Support serverless use case for independent static web hosting

### DISCUSSION THREAD

**Discussion by bkil** [asciinema/discussions#92] [Ideas]
In the tildeverse, you are allocated a little space that you can use over ssh/sftp/rsync and can serve your own small static web site. A relevant use case for it is a txt-based federated social networking https://github.com/buckket/twtxt

Similarly, it would be desirable if anybody residing on an instance could publish their own asciinema recordings on their own small "portal" and if one had the option to discover the recordings all over the world stored on such instances.

Some very small improvements would be needed for this.

### Create a config file entry to specify publishing targets

* ssh/sftp/rsync, user, host, remote folder
* the default uploading target, so it need not be specified every time through the CLI

### Implement mechanism through which the CLI could upload to our own publishing target as well

* the cast file
* the web player assets (if it is missing or needs updating),
* an HTML skeleton for playback of the given recording,
* an index HTML main page that lists all of our recordings (possibly one per category),
* a small Atom feed file of our recent recordings for our followers
* some metadata for "federation"/indexer use cases, like a CSV (or JSON, XML, Atom, etc.) that lists all of our recordings along with metadata, possibly one file per category (or tag), along with one that lists everything (related: asciinema/asciinema#513)
* list of other federated asciinema accounts who we publicly follow or just endorse (to aid discovery, similar to twtxt and web rings)

### Specify follower account to user-agent header

* similarly to twtxt, if we have our own federated asciinema account and watch the recording of someone else (or follow someone else), our client may (optionally) encode our own feed in the user-agent string so they could follow us back

### Discovery interface

A "portal" view as a simple web page and using ncurses for the CLI.

* to list, search, play and follow recordings and users
* our own
* those who we follow
* everyone in the global asciinema-verse (transitively reachable through users who we know of)
* uploaded on asciinema.org

**Comment by modem7** [comment]
Could also be quite useful for github pages.

**Comment by ku1ik** [comment]
I'm thinking about implementing configurable recording targets for asciinema 3.0 (in the future). Pretty much what you described: possibility to record to a directory (with template filename), upload via HTTP, or a custom upload script, with default encoded in config file :+1: This would also solve problem some people have with asciinema recorder defaulting to upload to asciinema.org. 

With option to use your own shell script for upload you could do anything you like, including upload over ssh, regenerating index.html and whatnot. You could even call hugo/jekyll/gatsby in such a script to regenerate whole "portal". I think this would be a good start.

I mentioned asciinema 3.0 - it's not planned yet, but major version bump would probably be a good idea for core behaviour change like not defaulting to asciinema.org (without initial consent/setup).
